### PR TITLE
Introduce `Config.from_json`

### PR DESCRIPTION
**Pull request by carmocca**
@rasbt This will fix the `condense_ratio` issue

### CODE REVIEW

File: lit_gpt/config.py [hunk 1]
```diff
@@ -83,6 +85,17 @@ def from_name(cls, name: str, **kwargs: Any) -> Self:
...
 
+    @classmethod
+    def from_json(cls, path: Union[str, Path], **kwargs: Any) -> Self:
+        with open(path, encoding="utf-8") as fp:
+            json_kwargs = json.load(fp)
+        if "condense_ratio" in json_kwargs:  # legacy name
+            json_kwargs["rope_condense_ratio"] = json_kwargs.pop("condense_ratio")
+        if "condense_ratio" in kwargs:  # legacy name
+            kwargs["rope_condense_ratio"] = kwargs.pop("condense_ratio")
+        json_kwargs.update(kwargs)
```
Comment (Andrei-Aksionov) [inline lit_gpt/config.py:96]: This one changes the previous behavior: before the default values were overridden by the values from the config file, and now it's the opposite.
In my books, values from the config file are more important.

File: tests/test_config.py [hunk 2]
```diff
@@ -12,3 +15,19 @@ def test_config():
 
     config = Config.from_name("pythia-70m", block_size=4096)
     assert config.block_size == 4096
+
+
+def test_legacy_args(tmp_path):
+    from lit_gpt import Config
+
+    config = Config.from_name("pythia-70m", condense_ratio=2)
+    assert config.rope_condense_ratio == 2
```
Comment (Andrei-Aksionov) [inline tests/test_config.py:24]: Maybe tests should also check that `condense_ratio` is no longer in the config?

File: lit_gpt/config.py (hunk 1 above)
Comment (carmocca) [reply lit_gpt/config.py:96]: We have cases for both: finetune/lora overrode the args whereas xla/generate/adapter overrode the config.

I think it's more likely that users add an arg to the source code than modify the config json, in that case I prefer the current behaviour

File: tests/test_config.py (hunk 2 above)
Comment (Andrei-Aksionov) [reply tests/test_config.py:24]: Ok :)

**Pull request merged**